python-lib/text_extraction/__init__.py

```diff
@@ -0,0 +1,54 @@
+import docx
+from io import BytesIO
+import logging
+import os
+import pypandoc
+import pypdfium2 as pdfium
+import tempfile
+
+
+logger = logging.getLogger(__name__)
+
+
+def try_download_pandoc():
+    """ download pandoc prebuilt binaries into the temporary job folder """
+    try:
+        import pypandoc
+        pandoc_tmp_directory = os.getcwd()
+        pypandoc.download_pandoc(targetfolder=pandoc_tmp_directory, download_folder=pandoc_tmp_directory)
+        return True
+    except Exception as e:
+        logger.warning("Failed to download pandoc binaries: {}".format(e))
+        return False
+
+
+def extract_text_content(file_bytes, extension, with_pandoc):
+    """
+    Extract content from file bytes:
+    - First try to extract with PDF or doc file
+    - Then try using pandoc to extract other files into plain text.
+    - Finally, just decode the bytes if pandoc failed or is not downloaded.
+    """
+    if extension == "pdf":
+        pdf_pages = pdfium.PdfDocument(file_bytes)
+        return "\n".join([page.get_textpage().get_text_range() for page in pdf_pages])
+    elif extension == "docx":
```

Comment:
Shall we add `doc` here too ?

Reply:
actually `.doc` files don't work (either with `docx` or `pandoc`), reading old `.doc` files requires having either microsoft word installed or having some specific system package (like `soffice` which is the program called by unstructured for `.doc` files: https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/common.py#L155)

Reply:
OK fair enough, shall we add a specific error message to suggest converting to docx instead ?